cmd/runner/main.go

```diff
@@ -198,22 +206,54 @@ func main() {
 	}
 
 	go (func() {
+		currentInterval := 1
+		errChan := make(chan error, concurrentWorkflows)
+		
 		for {
 			pool.Submit(func() {
 				wf, err := starter()
 				if err != nil {
-					log.Println("Unable to start workflow", err)
+					fmt.Fprintf(os.Stderr, "Unable to start workflow: %v\n", err)
+					errChan <- err
 					return
 				}
-
+				
 				if waitForCompletion {
 					err = wf.Get(context.Background(), nil)
 					if err != nil {
 						log.Println("Workflow failed", err)
+						errChan <- err
 						return
 					}
 				}
+				
+				errChan <- nil
 			})
+			
+			var lastErr error
+			drained := false
+			for {
+				select {
+				case err := <-errChan:
+					lastErr = err
+					drained = true
+				default:
+					goto checkError
+				}
+			}
+			
+		checkError:
+			if drained && lastErr != nil {
+				if !disableBackOff {
+					if currentInterval < maxInterval && maxInterval != 0 {
+						currentInterval *= factor
+					}
+					fmt.Fprintf(os.Stderr, "Waiting for %d seconds before retrying to start workflow...\n", currentInterval)
+					time.Sleep(time.Duration(currentInterval) * time.Second)
```

Comment:
We should bump `currentInterval` after sleeping rather than before. Otherwise we'll always jump straight to 2 seconds for the first sleep, which I don't think is the intent.